Community Guide (#296)
woodpecker-ci/woodpecker: CONTRIBUTING.md

# Contributing

## Maintainers

> To make sure every PR is checked, we have **team maintainers**.  

Reviewer: ```suggestion
To make sure every Pull Request (PR) is checked, we have **team maintainers**.  
```

Every PR **MUST** be reviewed by at least **one** maintainer (or owner) before it can get merged.  
A maintainer should be a contributor and contributed at least 4 accepted PRs.
A contributor should apply as a maintainer in the [Discord #develop](https://discord.gg/fcMQqSMXJy) channel.
The owners or the team maintainers may invite the contributor.
A maintainer should spend some time on code reviews.

If a maintainer has no time to do that, they should apply to leave the maintainers team and we will give them the honor of being a member of the [advisors
team](https://github.com/orgs/woodpecker-ci/teams/advisors/members).
Of course, if an advisor has time to code review, we will gladly welcome them back to the maintainers team.
If a maintainer is inactive for more than 3 months and forgets to leave the maintainers team, the owners may move him or her from the maintainers team to the advisors team.

> For security reasons, Maintainers should use 2FA for their accounts and if possible provide gpg signed commits.  

Reviewer: ```suggestion
For security reasons, Maintainers must use 2FA for their accounts and if possible provide GPG signed commits.  
```

MFA should be defacto these days.

https://help.github.com/articles/securing-your-account-with-two-factor-authentication-2fa/
https://help.github.com/articles/signing-commits-with-gpg/

## Owners

Since Woodpecker is a pure community organization without any company support,
to keep the development healthy we will elect two owners every year.  

> This can allso happen when a owner propose a vote or the majoirty of the maintainers do so.  

Reviewer: ```suggestion
This can also happen when a owner propose a vote or the majority of the maintainers do so.  
```

> All maintainers may vote to elect up to two candidates. When the new owners have been elected, the old owners will give up ownership to the newly elected owners.  

Reviewer: This feels a bit odd to me - while the idea in principal seems great what happens if the current owners do an amazing job at running the project and want to continue running? Is this common amongst OSS projects?

Author: Well this was discussed my **me** and **laszlocph**... he did propose a 6Mont periode (sine a lot can happen...) and I thought this is to frequent.
So we did a 1Jear as gitea do with the clause if something bad happen there is a clear way to go ...

In general if the owners do good - nothing will change in the vote ...

If woodpecker do have a wider developer base we could do 3 owner and other ajustments ... just do minimize a lock-out factor. For now we dont have that attenton and are not that big so we should go with that one.

cheers - @6543

Reviewer 2: If there is an exit close, I'm fine with 1y too.
